### PR TITLE
Add fetchMetrics API to BaseStatsReporter

### PR DESCRIPTION
**Pull request by majetideepak**
Return the aggregated metrics in a serialized string format.

### CODE REVIEW

**Comment by majetideepak** [comment]
@tanjialiang can you help review/land this change? Thanks!

**Review by xiaoxmeng — COMMENTED** [review]
@majetideepak why we need this? thanks!

**Comment by majetideepak** [comment]
@xiaoxmeng We need this to simplify the prometheus metrics implementation in Presto.
See @tanjialiang's comment here https://github.com/prestodb/presto/pull/22360#discussion_r1635316603.
Without this, we need to add a custom implementation something like.
```
std::string getSerializedMetrics() {
  if (auto reporter =
          std::dynamic_pointer_cast<prometheus::PrometheusStatsReporter>(
              folly::Singleton<velox::BaseStatsReporter>::try_get())) {
    return reporter->getSerializedMetrics();
  }
  VELOX_UNREACHABLE("No metrics reporter is enabled");
}
```

**Comment by majetideepak** [comment]
@xiaoxmeng, @tanjialiang any thoughts on this API? Thanks!

File: velox/common/base/StatsReporter.h [hunk 1]
```diff
@@ -135,6 +135,9 @@ class BaseStatsReporter {
   virtual void addHistogramMetricValue(folly::StringPiece key, size_t value)
       const = 0;
 
+  /// Return the metrics in a serialized string format.
```
Comment (tanjialiang) [inline velox/common/base/StatsReporter.h:138]: /// Return the aggregated metrics in a serialized string format.

Comment (tanjialiang) [reply velox/common/base/StatsReporter.h:138]: I'm wondering if this API should be const. Depending on implementations, clean-like operations can happen inside per each fetch interval.

**Review by tanjialiang — COMMENTED** [review]
Thanks for the change. Some minor issues.

File: velox/common/base/StatsReporter.h (hunk 1 above)
Comment (majetideepak) [reply velox/common/base/StatsReporter.h]: Makes sense to remove the const. I address the comments. Thanks!

**Review by tanjialiang — APPROVED** [review]

**Pull request closed without merging**